packages/theme/src/components/avatar.ts

```diff
@@ -23,18 +23,27 @@ const baseStyleExcessLabel: SystemStyleFunction = (props) => {
 }
 
 const baseStyleContainer: SystemStyleFunction = (props) => {
-  const { name, theme } = props
-  const bg = name ? randomColor({ string: name }) : "gray.400"
-  const isBgDark = isDark(bg)(theme)
+  const { theme } = props
 
-  let color = "white"
-  if (!isBgDark) color = "gray.800"
+  const colors = Object.keys(theme.colors).filter(
+    (c) =>
+      ![
+        "whiteAlpha",
+        "blackAlpha",
+        "white",
+        "black",
+        "transparent",
+        "current",
+      ].includes(c),
+  )
+
+  const bgColor = colors[Math.floor(Math.random() * colors.length)]
```

Comment:
We cannot use `Math.random` in code which is run multiple times, because this will return _random_  colors on each run.
This code will be run on each render of the Avatar component, so we need a stable algorithm.

I think the implementation of `randomColor` does only use tinycolor to create a fallback value, which we could either remove or replace.

Reply:
Good point. I'll check again on that :)

Reply:
I've added a custom fallback to replace the tinycolor implementation. The fallback is not used in our own code. https://github.com/chakra-ui/chakra-ui/blob/next/packages/theme/src/components/avatar.ts#L27
Not using the function without input, thus no case for the avatar.